### PR TITLE
[SPARK-36225][PYTHON][DOCS] Use DataFrame in python docstrings

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
Changing references to Dataset in python docstrings to DataFrame

### Why are the changes needed?
no Dataset class in pyspark

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Doc change only

### CODE REVIEW

File: python/pyspark/ml/util.py [hunk 1]
```diff
@@ -460,8 +461,8 @@ class DefaultParamsReadable(MLReadable):
     If a :py:class:`Params` class stores all data as :py:class:`Param` values,
     then extending this trait will provide a default implementation of reading saved
     instances of the class. This only handles simple :py:class:`Param` types;
-    e.g., it will not handle :py:class:`Dataset`. See :py:class:`DefaultParamsWritable`,
-    the counterpart to this trait.
+    e.g., it will not handle :py:class:`pyspark.sql.DataFrame`. See
+    :py:class:`DefaultParamsWritable`, the counterpart to this trait.
```
Comment (HyukjinKwon) [inline python/pyspark/ml/util.py:465]: I think we can just use `:class:`

**Comment by HyukjinKwon** [comment]
Nice, thanks for your contributions!

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise

File: python/pyspark/ml/util.py [hunk 2]
```diff
@@ -351,7 +351,8 @@ class DefaultParamsWritable(MLWritable):
     class stores all data as :py:class:`Param` values, then extending this trait will provide
     a default implementation of writing saved instances of the class.
     This only handles simple :py:class:`Param` types; e.g., it will not handle
-    :py:class:`Dataset`. See :py:class:`DefaultParamsReadable`, the counterpart to this trait.
+    :py:class:`pyspark.sql.DataFrame`. See :py:class:`DefaultParamsReadable`, the counterpart
+    to this trait.
```
Comment (HyukjinKwon) [inline python/pyspark/ml/util.py:355]:
```suggestion
    to this class.
```

File: python/pyspark/ml/util.py (hunk 1 above)
Comment (HyukjinKwon) [inline python/pyspark/ml/util.py:465]:
```suggestion
    :py:class:`DefaultParamsWritable`, the counterpart to this class.
```

Comment (dominikgehl) [reply python/pyspark/ml/util.py:465]: in this file, :py:class is used everywhere . I don't mind changing the above, but maybe we should then change it everywhere for consistency ?

Comment (HyukjinKwon) [reply python/pyspark/ml/util.py]: Yeah, let's just leave it. that's fine

**Comment by HyukjinKwon** [comment]
@dominikgehl can you check the build status? linter failed

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@dominikgehl mind taking a look for the linter failure?

**Comment by dominikgehl** [comment]
> @dominikgehl mind taking a look for the linter failure?

Done

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.2 and branch-3.1.

**Pull request closed without merging**